pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -360,6 +360,12 @@ func (c *Cluster) assignMons(mons []*monConfig) error {
 		return fmt.Errorf("failed to get available nodes for mons. %+v", err)
 	}
 
+	nodeCount := len(availableNodes)
+	monCount := len(mons)
+	if !c.spec.Mon.AllowMultiplePerNode && monCount > nodeCount {
+		return fmt.Errorf("the desired mon count is %d while the available node count is %d despite allowMultiplePerNode is %t", monCount, nodeCount, c.spec.Mon.AllowMultiplePerNode)
+	}
+
 	nodeIndex := 0
 	for _, m := range mons {
```

Comment:
@travisn I think the root cause is that mon would be forced to be scheduled as many as the number of monConfig inputted.
The available nodes count is not considered even if `allowMultiplePerNode: false`.